### ISSUE TITLE
Systemd is not notified when error occurs in main scrip

### ISSUE THREAD

**Issue by printercu** [tarantool/tarantool#4382]
Tarantool version:
master

Bug description:
This catch branches https://github.com/tarantool/tarantool/blob/260f63283deff986395d38636978bb6cf90ae4ff/src/main.cc#L851-L859
are not executed when error occurs in initial lua script, so systemd_snotify is not invoked.

Steps to reproduce:
1) Add `say_crit` to catch branches
2) run any invalid lua script.

**Comment by Totktonada** [comment]
@Kasen points that it may be connected with `RateLimitInterval` journald option (which default is 10000 messages in 30s).

**Comment by printercu** [comment]
I don't think so. As described in issue the code in `catch` block is never reached, even in setups without systemd.

**Comment by Gerold103** [comment]
This has nothing to do with systemd or any other way of logging. This is because 1) main script works in a separate fiber, so its exceptions can't reach this try-catch block, 2) in the separate fiber it does not throw exceptions either. On this line https://github.com/tarantool/tarantool/blob/master/src/lua/init.c#L546 it does panic instead.
I will make a patch which will eliminate panics and will use diag_set + diag_raise (I already made it, actually, but perhaps it won't pass review).
